app/models/kaui/allowed_user.rb

```diff
@@ -11,6 +11,9 @@ class AllowedUser < ApplicationRecord
 
     # Create the user locally and in Kill Bill (if needed)
     def create_in_kb!(password, roles = [], user = nil, reason = nil, comment = nil, options = {})
+      # only save locally if is managed externally (ex. LDAP, shiro ext)
```

Comment:
I think there is some confusion here:

* Managed externally means outside of Kill Bill (LDAP or Okta). Kill Bill doesn't know about the password
* Managed _internally_ means inside Kill Bill (shiro or JDBC). Kill Bill does know about the password, however only the JDBC password is updatable.

So, in summary, only the users returned by `getUserRoles` can have their password updated (see description in ticket).

Given this, I'm also not sure you need to persist that information in the Kaui DB.

Reply:
@pierre, I will post an update to the code to make thing clear, but the thing is if I do not persist the `managed externally` flag, then I will not be able to distinguish between `managed externally` and `shiro`. At the moment of editing a user the checkbox will never be selected since there are no way of differentiate between `shiro` and `managed externally`.

Reply:
I overlooked the behavior of `getUserRoles` which always returns a `UserRolesJson`. I'm wondering if we shouldn't change the implementation and return `404` if the user doesn't exist.

In `public List<UserRolesModelDao> getUserRoles(final String username)`, we could add:

```
final UserModelDao userModelDao = usersSqlDao.getByUsername(username);
if (userModelDao == null) {
  throw new SecurityApiException(ErrorCode.SECURITY_INVALID_USER, username);
}
```

and catch the exception at the JAX-RS level to return `404`. If `404`, the user row doesn't exist, hence the password shouldn't be editable.

Would this solve the issue and not require the DB changes? If so, can you make the changes on the Kill Bill side as well?